CONTRIBUTING.md

```diff
@@ -0,0 +1,900 @@
+
+Always surround any other binary operator with whitespace, including assignment operators. Add line breaks after the operator, not before.
+
+```js
+// Good
+var a = 1 + 2
+a = 3
+a += 4
+
+var a = 1 + 2 + 3 + 4 + 5 +
+	6 + 7 + 8 + 9 + 10
+
+// Bad
+var a = 1+2
+var a = 1 +2
+var a = 1+ 2
+var a=1+2
+var a=1 + 2
+a=3
+a+=4
+
+var a = 1 + 2 + 3 + 4 + 5
+	+ 6 + 7 + 8 + 9 + 10
+```
+
+As an exception, casting to a 32-bit integer (i.e. `x|0`) doesn't require whitespace surrounding it.
+
+```js
+// This is okay
+var casted = value|0
+```
+
+### Unary operators
+
+Always use whitespace between an unary keyword operator (e.g. `delete`, `new`) with whitespace.
+
+Do not use spaces between any other unary operator and the value they apply to.
+
+```js
+// Good
+!a
+!!a
+~~a
+a++
+++a
+-1
+~1
+
+// Okay
++!a
+
+// Bad
+! a
+!! a
+! ! a
+++ a
+a ++
++ ! a
+
+// Even worse
+- 1
+~ 1
+```
+
+Don't use `new function () {}`. Just use an object literal instead, and an IIFE if needed.
+
+## Objects
+
+### Exterior whitespace
+
+Do not include space between the opening curly brace and the first object key in single line objects. For multi-line objects, don't include any extra preceding whitespace at the beginning or end.
+
+```js
+// Good
+var object = {foo: 1}
+
+var object = {
+	foo: 1,
+	bar: 2,
+	baz: 3,
+	quux: "string",
+}
+
+// Bad
+var object = { foo: 1 }
+
+var object = {
+
+	foo: 1,
+	bar: 2,
+	baz: 3,
+	quux: "string",
+
+}
+```
+
+### Interior whitespace
+
+Use no space before an object key and the colon, but use a single space between the colon and value.
+
+```js
+// Good
+var object = {foo: 1}
+
+// Bad
+var object = {foo:1}
+var object = {foo : 1}
+
+// Very bad
+var object = {foo :1}
+```
+
+### Larger objects
+
+Non-trivial objects should occupy multiple lines.
+
+```js
+// Good
+var object = {
+	foo: 1,
+	bar: 2,
+	baz: 3,
+	quux: "string",
+}
+
+// Bad
+var object = {foo: 1, bar: 2, baz: 3, quux: "string"}
+```
+
+### Comma placement
+
+Commas come last. Also, use trailing commas if the object takes multiple lines.
+
+```js
+// Good
+var object = {
+	foo: 1,
+	bar: 2,
+}
+
+// Bad
+var object = {
+	foo: 1,
+	bar: 2 // No trailing comma
+}
+
+var object = {
+	foo: 1
+,	bar: 2 // Comma first
+}
+
+// Trailing comma in single-line object
+var object = {foo: 1,}
+```
+
+### Member access
+
+When chaining methods and/or properties across multiple lines, dots come first.
+
+```js
+// Good
+object
+	.foo()
+	.bar()
+
+// Bad
+object.
+	foo().
+	bar()
+```
+
+### Property quoting
+
+Quote properties when needed. Never quote valid identifiers. This may lead to some inconsistency in whether properties are quoted or not in the object, but that inconsistency is okay, and is the only exception in consistency.
+
+```js
+// Good
+var object = {
+	foo: 1,
+	bar: 2,
+}
+
+var object = {
+	foo: 1,
+	"non-identifier": 2,
+}
+
+// Bad
+var object = {
+	"foo": 1,
+	"bar": 2,
+}
+
+var object = {
+	"foo": 1,
+	"non-identifier": 2,
+}
+```
```

Comment:
Only in IE8 and before. And this already wouldn't run in it without an enormous amount of polyfills.

Also, FWIW, Microsoft is planning on dropping support in about 2 months for 99.99% of its users (only older versions of embedded Windows could be supported). It's not the latest version available for any supported OS that's not specifically for embedded applications.

I think we can drop support for it.